src/System.Net.Http/tests/FunctionalTests/ResponseStreamTest.cs

```diff
@@ -68,6 +69,26 @@ public ResponseStreamTest(ITestOutputHelper output)
             }
         }
 
+        [Fact]
+        public async Task ReadAsStreamAsync_Cancel_TaskIsCanceled()
+        {
+            var cts = new CancellationTokenSource();
+
+            using (var client = new HttpClient())
+            {
+                HttpResponseMessage response =
+                    await client.GetAsync(HttpTestServers.RemoteEchoServer, HttpCompletionOption.ResponseHeadersRead);
+                Stream stream = await response.Content.ReadAsStreamAsync();
+
+                var buffer = new byte[2048];
+                Task task = stream.ReadAsync(buffer, 0, buffer.Length, cts.Token);
```

Comment:
Yes, theoretically, there is a race condition.  I pondered that as I was adding this test to the CI.  Without the fix, we hang as the task never completes.  I could modify the check and validate that the task moves to either a Canceled or Completed (successfully) state

The other way to test this is by adding a UnitTest to WinHttpHandler specifically and mocking the network layer to get the timings exact.  But I thought it was more valuable to have an HttpClientHandler test to help exercise CurlHandler as well.